src/wp-includes/class-wp-scripts.php

```diff
@@ -464,7 +513,21 @@ public function print_inline_script( $handle, $position = 'after', $display = tr
 		$output = trim( implode( "\n", $output ), "\n" );
 
 		if ( $display ) {
-			printf( "<script%s id='%s-js-%s'>\n%s\n</script>\n", $this->type_attr, esc_attr( $handle ), esc_attr( $position ), $output );
+			if ( 'after-non-standalone' === $position ) {
+				$initial_type_attr = $this->type_attr;
+				$this->type_attr   = " type='text/template'";
+				printf(
+					'<script%1$s id=\'%2$s-js-after\' data-wp-executes-after=\'%2$s\'>%5$s%4$s%5$s</script>%5$s',
+					$this->type_attr,
+					esc_attr( $handle ),
+					esc_attr( $position ),
+					$output,
+					PHP_EOL
+				);
+				$this->type_attr = $initial_type_attr;
+			} else {
+				printf( "<script%s id='%s-js-%s'>\n%s\n</script>\n", $this->type_attr, esc_attr( $handle ), esc_attr( $position ), $output );
```

Comment:
Numbered placeholders and readability:

```suggestion
				printf(
					"<script%1$s id='%2$s-js-%3$s'>\n%4$s\n</script>\n",
					 $this->type_attr,
					 esc_attr( $handle ),
					 esc_attr( $position ),
					 $output
				);
```

Reply:
Reply same as above (first comment)
https://github.com/10up/wordpress-develop/pull/54/files#r1168320548